app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,40 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
```

Comment:
I know this is not marked as required but adding some further info may help to clarify. Something like "Leaving UUID blank will result in one being auto created for this host." or "Setting a UUID allows for connecting a host with subscription-manager later."